scenario/state.py

```diff
@@ -207,25 +204,28 @@ def deferred(self, handler: Callable, event_id: int = 1) -> "DeferredEvent":
         return self().deferred(handler=handler, event_id=event_id)
 
 
-def _generate_new_relation_id():
-    global _RELATION_IDS_CTR
-    _RELATION_IDS_CTR += 1
-    logger.info(
-        f"relation ID unset; automatically assigning {_RELATION_IDS_CTR}. "
-        f"If there are problems, pass one manually.",
-    )
-    return _RELATION_IDS_CTR
-
-
 @dataclasses.dataclass(frozen=True)
 class RelationBase(_DCBase):
     endpoint: str
 
     # we can derive this from the charm's metadata
     interface: str = None
 
+    NEXT_RELATION_ID = 1
```

Comment:
I guess the all-caps is to distinguish it from the dataclass fields? But what actually makes it a non-dataclass-field is that it doesn't have a type, correct?